sherpa/astro/io/__init__.py

```diff
@@ -948,6 +1066,463 @@ def convert(column, dtype):
+        HDUCLASS = 'OGIP' - file format is OGIP standard.
+        HDUCLAS1 = 'RESPONSE' - extension contains response data.
+        HDUCLAS2 = 'RSP_MATRIX' - extension contains a response matrix.
+        HDUVERS = '1.3.0' - version of the file format.
+        TLMIN# - the first channel in the response.
+
+    These are optional but mechanical so can be added:
+
+        NUMGRP - sum(N_GRP)
+        NUMELT - sum(N_CHAN)
+
+    These are optional:
+
+        LO_THRES - minimum probability threshold
+        HDUCLAS3 - 'REDIST', 'DETECTOR', 'FULL'
+
+    For the EBOUNDS extension we have:
+
+        EXTNAME  = 'EBOUNDS'
+        TELESCOP
+        INSTRUME
+        FILTER
+        CHANTYPE
+        DETCHANS
+        HDUCLASS = 'OGIP'
+        HDUCLAS1 = 'RESPONSE'
+        HDUCLAS2 = 'EBOUNDS'
+        HDUVERS = '1.2.0'
+
+    We do not add the RMFVERSN, HDUVERS1, or HDUVERS2 keywords which
+    are marked as obsolete.
+
+    """
+
+    # The UI layer wraps up a DataRMF into RMF1D quite often, so
+    # support both types as input.
+    #
+    if not isinstance(dataset, (DataRMF, RMF1D)):
+        raise IOErr("data set is not a RMF")
+
+    rmfdata = _reconstruct_rmf(dataset)
+
+    # The default keywords; these wil be over-ridden by
+    # anything set by the input.
+    #
+    default_matrix_header = {
+        "EXTNAME": "MATRIX",
+        "HDUCLASS": "OGIP",
+        "HDUCLAS1": "RESPONSE",
+        "HDUCLAS2": "RSP_MATRIX",
+        "HDUVERS": "1.3.0",
+        "TELESCOP": "none",
+        "INSTRUME": "none",
+        "FILTER": "none",
+        "CHANTYPE": "PI",
+        "DETCHANS": "none",
+        "NUMGRP": 0,
+        "NUMELT": 0
+    }
+
+    default_ebounds_header = {
+        "EXTNAME": "EBOUNDS",
+        "HDUCLASS": "OGIP",
+        "HDUCLAS1": "RESPONSE",
+        "HDUCLAS2": "EBOUNDS",
+        "HDUVERS": "1.2.0",
+        "TELESCOP": "none",
+        "INSTRUME": "none",
+        "FILTER": "none",
+        "CHANTYPE": "PI",
+        "DETCHANS": "none"
+    }
+
+    # Header Keys
+    header = dataset.header
+
+    # Merge the keywords (at least for the MATRIX block).
+    #
+    matrix_header = {**default_matrix_header, **header}
+    ebounds_header = {**default_ebounds_header}
+
+    matrix_header["NUMGRP"] = rmfdata["NUMGRP"]
+    matrix_header["NUMELT"] = rmfdata["NUMELT"]
+    matrix_header["DETCHANS"] = dataset.detchans
+
+    # Copy values over.
+    #
+    for copykey in ["TELESCOP", "INSTRUME", "FILTER", "CHANTYPE",
+                    "DETCHANS"]:
+        ebounds_header[copykey] = matrix_header[copykey]
+
+    # The column ordering for the ouput file is determined by the
+    # order the keys are added to the data dict.
+    #
+    # To allow the backend convert to Variable-Length fields, we
+    # have F_CHAN/N_CHAN/MATRIX either be a 2D ndarray (so fixed
+    # output) OR a list of rows (use VLF). It's not ideal: we could
+    # wrap up in a local VLF type just to indicate this to the
+    # backend.
+    #
```

Comment:
But VLF is also optional. The standard even says that you might want to use rectangular matrixes instead of VLF, even is the rectangualr matrix is bigger, because it's much faster to read than VLF. So, don't worry too much about what gets to be a VLF and what gets to be an array.